game/state/city/vehicle.cpp

```diff
@@ -2184,36 +2205,37 @@ void Vehicle::update(GameState &state, unsigned int ticks)
 								// Try to attack. Attack won't happen if enemy is out of range.
 								attackTarget(state, enemy);
 							}
-							if (IsIdle)
+							// Search for closest target to face towards (if not already rotating
+							// for some reason)
+							if (IsIdle && this->angularVelocity == 0.0f)
 							{
 								// If not moving anywhere then search for closest target around.
 								enemy = findClosestEnemy(state, tileObject, {8, 8});
 								if (enemy)
 								{
 									// Determine target position.
 									Vec3<float> enemypos = enemy->getPosition();
-									auto facing_direction = type->directionToVector(direction);
-									Vec2<float> facing2d = {facing_direction.x, facing_direction.y};
-									Vec2<float> target2d = {enemypos.x - position.x,
-									                        enemypos.y - position.y};
-									facing2d = glm::normalize(facing2d);
-									target2d = glm::normalize(target2d);
+									Vec2<float> facing2d = facingToXY(this->facing);
+									Vec2<float> target2d = glm::normalize(Vec2<float>{
+									    enemypos.x - position.x, enemypos.y - position.y});
 									float angleXY = glm::angle(facing2d, target2d);
 									// Check if target is in our firing arc.
 									if (angleXY > (float)arc.x * (float)M_PI / 8.0f)
 									{
-										// Turn towards closest target (firing will start
-										// automatically).
-										// It is pseudoscalar product. It is needed to determine
-										// direction: right or left.
-										if (facing2d.x * target2d.y - facing2d.y * target2d.x > 0)
+										this->goalFacing = xyToFacing(target2d);
+										float d = facingDistance(this->goalFacing, this->facing);
+										// TODO: Should this nudge CCW/CW for animation purposes?
+										if (d > 0.0f)
 										{
-											this->facing += 0.06f * (float)M_PI * (float)ticks;
+											// Rotate CW towards target
+											this->angularVelocity = this->getAngularSpeed();
```

Comment:
Would this overshoot if it's less than getAngularSpeed() away? Should we clamp this to to min(d, getAngularSpeed())?

Reply:
I'm not entirely sure TBH. I think this will cause the ticksToTurn to be 0, so the VehicleMover will snap the vehicle rotation to the goalFacing value. I've implemented this anyways because I think it's better to make this case immediately apparent rather than to rely on that convoluted logic